### PR TITLE
MEN-1806: print artifact name from the cli 

### PR DESCRIPTION
**Pull request by oleorhagen**
Changelog: Title

Signed-off-by: Ole Petter <ole.orhagen@cfengine.com>

### CODE REVIEW

**Comment by estenberg** [comment]
That was fast. :)
Not sure about the option name, "current-artifact", though, it feels a bit nonstandard and cumbersome.
Maybe use some inspiration from other Unix tools or package managers?

How about -show-artifact?

@kacf @pasinskim any opinions?

File: main.go [hunk 1]
```diff
@@ -271,6 +276,15 @@ func ShowVersion() {
 	os.Stdout.Write([]byte(v))
 }
 
+func PrintArtifactName() {
+	aname, err := ioutil.ReadFile("/etc/mender/artifact_info")
+	if err != nil {
+		fmt.Printf("failed to read artifact name: err: %v", err)
+		return
+	}
+	fmt.Println(aname)
```
Comment (kacf) [inline main.go]: This is the format of the file:

```
artifact_name=mender-image-1.0
```

It should only print the value, not the key. And you should assume that there may be other fields in there in the future.

File: main.go [hunk 2]
```diff
@@ -103,6 +105,8 @@ func argsParse(args []string) (runOptionsType, error) {
 
 	bootstrap := parsing.Bool("bootstrap", false, "Perform bootstrap and exit.")
 
+	curArtifact := parsing.Bool("current-artifact", false, "print the current artifact name to the command line and exit")
```
Comment (kacf) [inline main.go]: Agree with Eystein, I think `show-artifact` is better than `current-artifact`.

**Comment by oleorhagen** [comment]
fixed!

File: main.go [hunk 3]
```diff
@@ -271,6 +278,25 @@ func ShowVersion() {
...
+	for s.Scan() {
+		if strings.HasPrefix(s.Text(), "artifact_name=") {
+			a := strings.Split(s.Text(), "=")
+			if len(a) == 2 {
+				fmt.Println(a[1])
+			} else {
+				fmt.Println("Wrong formatting of the artifact_info file")
+			}
+		}
+	}
```
Comment (kacf) [inline main.go]: There are a couple of corner cases here that I think there should produce an error:

* There is no `artifact_name=`
* There is more than one `artifact_name=`

Both can have severe consequences if they are wrong, so I think we should return error in these cases.

**Comment by oleorhagen** [comment]
another fixup

**Comment by kacf** [comment]
@pasinskim: Maybe we should raise the cyclomatic complexity requirement. `doMain` is not even two pages, and it still complains.

**Comment by pasinskim** [comment]
@kacf I would rather not to. It forces us to keep the code simple and is set the same in all places.

**Comment by kacf** [comment]
Ok @oleorhagen, you heard the man/ :-P

**Comment by oleorhagen** [comment]
Do not merge yet

**Comment by oleorhagen** [comment]
Now we should be good :)

File: main.go [hunk 4]
```diff
@@ -391,8 +422,19 @@ func doMain(args []string) error {
...
+
+func handleCLIOptions(runOptions runOptionsType, env *uBootEnv, device *device, config *menderConfig) error {
+
 	switch {
 
+	case *runOptions.version:
+		ShowVersion()
+		return nil
+	case *runOptions.showArtifact:
+		return PrintArtifactName("/etc/mender/artifact_info")
```
Comment (kacf) [inline main.go]: Oh, can you use `getConfDirPath()` here instead of hardcoding `/etc/mender`? Sorry, I forgot about it earlier.

Comment (oleorhagen) [reply main.go]: sounds reasonable :) fixed!

**Comment by GregorioDiStefano** [comment]
Your PR is based off tag: 1.3.0b1, hence why it keeps failing. 
Rebase your PR off master.

**Comment by estenberg** [comment]
Is this done? Task still open: https://tracker.mender.io/browse/MEN-1806

**Pull request merged**